test/index.test.ts

```diff
@@ -647,73 +678,17 @@ describe('HipThrusTS', () => {
 
         const pipedRightOnly = HTPipe({}, rightPartiallyCovered);
 
-        interface CorrectParam {
-          b: number;
-          other: string;
-        }
-        interface CorrectReturnValue {
-          c: number;
-        }
-        type ParamAssignableToCorrect = CorrectParam extends Parameters<
-          typeof pipedRightOnly.attachData
-        >[0]
-          ? true
-          : false;
-        type ParamAssignableFromCorrect = Parameters<
-          typeof pipedRightOnly.attachData
-        >[0] extends CorrectParam
-          ? true
-          : false;
-        type ReturnAssignableToCorrect = CorrectReturnValue extends ReturnType<
-          typeof pipedRightOnly.attachData
-        >
-          ? true
-          : false;
-        type ReturnAssignableFromCorrect = ReturnType<
-          typeof pipedRightOnly.attachData
-        > extends CorrectReturnValue
-          ? true
-          : false;
-        // @ts-expect-error
-        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
-        // @ts-expect-error
-        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
-        // @ts-expect-error
-        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
-        // @ts-expect-error
-        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
-
-        const pipedRightOnlyAttachDataReturnValues = await pipedRightOnly.attachData(
-          { b: bPassedIn, other: otherPassedIn }
+        await HTPipeTest(
+          pipedRightOnly,
+          'attachData',
+          { b: bPassedIn, other: otherPassedIn },
+          { c: cReturned },
+          true
         );
-
-        expect(pipedRightOnlyAttachDataReturnValues).to.has.property('c');
-        expect(pipedRightOnlyAttachDataReturnValues.c).to.be.equal(cReturned);
       });
-      it('no attaches data when left and right is empty objects', () => {
+      it('no attaches data when left and right is empty objects', async () => {
         const pipedWithEmptyObjectsOnly = HTPipe({}, {});
 
```

Comment:
aren't we losing type coverage in this corner case?
- [ ] revive manual type coverage here

Reply:
revived